tests/Microsoft.Identity.Test.LabInfrastructure/KeyVaultSecretsProvider.cs

```diff
@@ -144,19 +146,37 @@ private async Task<string> AuthenticationCallbackAsync(string authority, string
                             "Test setup error - cannot find a certificate in the My store for KeyVault. This is available for Microsoft employees only.");
                     }
 
-                    app = ConfidentialClientApplicationBuilder.Create(_config.ClientId)
+                    confidentialApp = ConfidentialClientApplicationBuilder.Create(_config.ClientId)
                                               .WithAuthority(new Uri(authority), true)
                                               .WithCertificate(cert)
                                               .Build();
 
-                    authResult = await app.AcquireTokenForClientAsync(scopes).ConfigureAwait(false);
+                    authResult = await confidentialApp.AcquireTokenForClientAsync(scopes).ConfigureAwait(false);
                     break;
                 case KeyVaultAuthenticationType.ClientSecret:
-                    app = ConfidentialClientApplicationBuilder.Create(_config.ClientId)
+                    confidentialApp = ConfidentialClientApplicationBuilder.Create(_config.ClientId)
                                               .WithAuthority(new Uri(authority), true)
                                               .WithClientSecret(_config.KeyVaultSecret)
                                               .Build();
-                    authResult = await app.AcquireTokenForClientAsync(scopes).ConfigureAwait(false);
+                    authResult = await confidentialApp.AcquireTokenForClientAsync(scopes).ConfigureAwait(false);
+                    break;
+                case KeyVaultAuthenticationType.UserCredential:
+                    publicApp = PublicClientApplicationBuilder.Create(_config.ClientId)
+                                                .WithAuthority(new Uri(authority), true)
+                                                .Build();
+                    try
+                    {
+                        authResult = await publicApp.AcquireTokenByIntegratedWindowsAuthAsync(scopes).ConfigureAwait(false);
+                    }
+                    catch (MsalUiRequiredException ex)
+                    {
+                        publicApp = PublicClientApplicationBuilder.Create(_config.ClientId)
+                                                .WithAuthority(new Uri(authority), true)
+                                                .WithClaims(ex.Claims)
+                                                .Build();
+
+                        authResult = await publicApp.AcquireTokenAsync(scopes).ConfigureAwait(false);
+                    }
```

Comment:
Don't forget to port these changes to ADAL test project.

Reply:
makes sense to at least add they IWA stuff on ADAL, however it may also be ok to keep using ADAL there as that is an additional test that things works for the actual library being developed?

---
In reply to: [254737016](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/868#discussion_r254737016) [](ancestors = 254737016)